packages/ann/ann/c_src/logistic_actf_component.cc

```diff
@@ -58,4 +58,63 @@ namespace ANN {
     buffer.printf("ann.components.actf.logistic{ name='%s' }", name.c_str());
     return buffer.to_string(buffer_list::NULL_TERMINATED);
   }
+
+  SparseLogisticActfANNComponent::SparseLogisticActfANNComponent(const char *name,
+        float beta, float avg_act):LogisticActfANNComponent(name) {
+      this->beta = beta; this->avg_act = avg_act; } 
+
+  ANNComponent *SparseLogisticActfANNComponent::clone() {
+    SparseLogisticActfANNComponent *obj = new SparseLogisticActfANNComponent(name.c_str(), beta, avg_act);
+    return obj;
+  }
+
+  char *SparseLogisticActfANNComponent::toLuaString() {
+    buffer_list buffer;
+    buffer.printf("ann.components.actf.sparse_logistic{ name='%s', beta=%f, rho=%f  }", name.c_str(), this->beta, this->avg_act);
+    return buffer.to_string(buffer_list::NULL_TERMINATED);
+  }
+ 
+  void SparseLogisticActfANNComponent::multiplyDerivatives(MatrixFloat *input_units,
+						     MatrixFloat *output_units,
+						     MatrixFloat *input_errors,
+						     MatrixFloat *output_errors) {
+    UNUSED_VARIABLE(input_units);
+    Kernels::applyLogisticDerivative(output_errors, output_units);
+
+    float eps = 1e-6;
+    // Computing average activations
+    MatrixFloat *current_avg = matSum(output_units, 0);
+    matScal(current_avg, 1.0f/static_cast<float>(output_units->getDimSize(0)));
+     
+    // -p/\hat{p} + (1-p)/(1-\hat{p}
+    // -p/\hat{p}
+    
+    MatrixFloat *sparse_errors = current_avg->clone();
+    matScalarAdd(sparse_errors, eps);
+    matDiv(sparse_errors, -avg_act);
+    
+    // (1-p)/(1-\hat{p})
+    MatrixFloat *aux = current_avg->clone();
+    matComplement(aux);
+    matScalarAdd(aux, eps);
+    matDiv(aux, (1.0f-avg_act));
+    matAddition(sparse_errors, aux, sparse_errors);
+
+    matAbs(sparse_errors, sparse_errors);
```

Comment:
`matAbs` is supossed to be **in-place** if no `dest` argument is given, so better `matAbs(sparse_errors)`